client/landing/stepper/declarative-flow/internals/components/user/index.tsx

```diff
@@ -80,7 +80,7 @@ const UserStep: Step = function UserStep( props ) {
 			isFullLayout
 			isLargeSkipLayout={ false }
 			hideBack
-			stepContent={ <StepContent { ...props } /> }
+			stepContent={ <StepContent flow={ flow } /> }
```

Comment:
Is the `flow` prop the only thing we need here?
Tested with `__experimentalUseBuiltinAuth` as `true` and `false` and didn't find any issues.